### PR TITLE
Implementing e2e tests kube - istio

### PR DESCRIPTION
**Pull request by M4tteoP**
Implements #3

### CODE REVIEW

File: kind-istio/workload/deploy-nginx.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+##################################################################################################
+# nginx dummy deployment
```
Comment (jcchavezs) [inline kind-istio/workload/deploy-nginx.yaml:2]: Quick question: why do we need this?

Comment (M4tteoP) [reply kind-istio/workload/deploy-nginx.yaml:2]: The wasm module is deployed on the proxy of the nginx workload and the e2e tests will try to connect to this nginx. The alternative could be deploy the waf just in front of the ingress and then making a request that would go nowhere maybe? I think is okay to have a service that we are trying to reach/protect

File: kind-istio/kind-config.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: kind.x-k8s.io/v1alpha4
```
Comment (jcchavezs) [inline kind-istio/kind-config.yaml:3]: Is this copied/inspired from somewhere? If that is the case please drop a line with the link

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -63,13 +88,87 @@ jobs:
             bin/
           key: wasm-module-build-${{ hashFiles('modsecurity-wasm-filter/**') }}
 
-      - name: "Spin up server and envoy"
+      - name: "Spin up local server and envoy"
         shell: bash
         run: |
           docker run --rm -d -p 8000:10080 jcchavezs/httpmole -response-status=200 &
```
Comment (jcchavezs) [inline .github/workflows/tests.yml]: This should be rebased.

File: kind-istio/kind-config.yaml (hunk 2 above)
Comment (M4tteoP) [reply kind-istio/kind-config.yaml:3]: It has been quite a lot revisited, but I added both the doc and file I used to do it

File: .github/workflows/tests.yml (hunk 3 above)
Comment (M4tteoP) [reply .github/workflows/tests.yml]: Rebased, let's see if tests are going to be okay. 
I also removed ` docker run --rm -d -p 8000:10080 jcchavezs/httpmole -response-status=200 &` being useless now

File: tests.sh [hunk 4]
```diff
@@ -1,11 +1,12 @@
 #!/bin/bash
 
+# if env variables are a valid alternative, local ones can be removed
 step=1
 total_steps=3
 max_retries=10 #seconds for the server reachability timeout
-application_url="http://localhost:8000"
-envoy_url_unfiltered="http://localhost:8001/home"
-envoy_url_filtered="http://localhost:8001/admin"
+application_url=$APPLICATION_URL
```
Comment (jcchavezs) [inline tests.sh]: How about making it optional with default values being the localhost ones?

**Pull request closed without merging**